Linux and64 binariy failed since v5.1.0

### Paste the command

`igir`

### Describe the bug

Running the [binary release 5.1.0  for Linux 64](https://github.com/emmercm/igir/releases/download/v5.1.0/igir-5.1.0-Linux-amd64.tar.gz) failed with
```
$ igir-5.1.0 
error: Cannot find module 'yargs/package.json' from '/$bunfs/root/index.js'

Bun v1.3.14 (Linux x64 baseline)
 
``` 

### Expected behavior

For the program to run standalone without an error.

### Debug log

[debug.log](https://github.com/user-attachments/files/28787431/debug.log)

### DAT(s) used

N/A

### igir version

v5.1.0

### Node.js version

N/A

### Operating system

Debian GNU/Linux 13 (trixie) AMD64

### Additional context

No debug log available as the error happens before any igir command is executed.

--- a/.github/workflows/node-compile.yml
+++ b/.github/workflows/node-compile.yml
@@ -143,6 +143,7 @@ jobs:
             # Compile the executable and test it before compressing it
             npm ci --foreground-scripts
             npm run compile -- . igir
+            rm -rf node_modules
             ./igir --help
             ./igir copy extract test clean report --dat test/fixtures/dats --input test/fixtures/roms --patch test/fixtures/patches --output e2e-extract
             ./igir copy zip test clean report --dat test/fixtures/dats --input test/fixtures/roms --patch test/fixtures/patches --output e2e-zip
@@ -168,6 +169,7 @@ jobs:
           # Package the executable and test it before compressing it
           npm ci --foreground-scripts
           npm run compile -- . igir
+          rm -rf node_modules
           ./igir --help
           ./igir copy extract test clean report --dat test/fixtures/dats --input test/fixtures/roms --patch test/fixtures/patches --output e2e-extract
           ./igir copy zip test clean report --dat test/fixtures/dats --input test/fixtures/roms --patch test/fixtures/patches --output e2e-zip
@@ -188,11 +190,10 @@ jobs:
           $BIN_OUTPUT="igir-$NPM_PKG_VERSION-${{ runner.os }}-${NODE_ARCH}.zip"
           echo "BIN_OUTPUT=${BIN_OUTPUT}" | Out-File -FilePath $env:GITHUB_ENV -Append # no need for -Encoding utf8
 
-          # Install dependencies...
-          npm ci --foreground-scripts
-
           # Package the executable and test it before compressing it
+          npm ci --foreground-scripts
           npm run compile -- . igir.exe
+          Remove-Item node_modules -Recurse -Force
           .\igir.exe --help
           .\igir.exe copy extract test clean report --dat test\fixtures\dats --input test\fixtures\roms --patch test\fixtures\patches --output e2e-extract
           .\igir.exe copy zip test clean report --dat test\fixtures\dats --input test\fixtures\roms --patch test\fixtures\patches --output e2e-zip

--- a/.github/workflows/node-test.yml
+++ b/.github/workflows/node-test.yml
@@ -40,6 +40,7 @@ jobs:
               - 'packages/**'
               - 'src/**'
               - 'test/**'
+              - '**/*.ts'
               - '*'
             should-delete-prebuilds:
               - '.github/workflows/node-addon-prebuild.yml'

--- a/.github/workflows/node-test.yml
+++ b/.github/workflows/node-test.yml
@@ -223,4 +223,4 @@ jobs:
       - uses: re-actors/alls-green@05ac9388f0aebcb5727afa17fcccfecd6f8ec5fe # release/v1
         with:
           jobs: ${{ toJSON(needs) }}
-          allowed-skips: node-lint, node-unit, bun-unit, node-e2e
+          allowed-skips: node-lint, node-unit, bun-unit, node-e2e, node-license

--- a/scripts/compile.ts
+++ b/scripts/compile.ts
@@ -1,7 +1,9 @@
 /// <reference types="@types/bun" />
 
 import child_process from 'node:child_process';
+import fs from 'node:fs';
 import module from 'node:module';
+import os from 'node:os';
 import path from 'node:path';
 
 import fg from 'fast-glob';
@@ -12,6 +14,7 @@ import Logger from '../src/console/logger.js';
 import { LogLevel } from '../src/console/logLevel.js';
 import IgirException from '../src/exceptions/igirException.js';
 import Package from '../src/globals/package.js';
+import Temp from '../src/globals/temp.js';
 import FsUtil from '../src/utils/fsUtil.js';
 
 const logger = new Logger(LogLevel.TRACE, process.stdout);
@@ -108,18 +111,24 @@ const bunBuildConfig = {
           );
 
           // Detect `const NAME = path.dirname(require.resolve('PKG/package.json'))`
-          // declarations and resolve each PKG to its on-disk root directory.
-          // Source files use this pattern when they need to reach internal
-          // files that the package's `exports` map doesn't expose; the next
-          // pass uses these roots to statically rewrite `require(`${NAME}/...`)`
-          // calls into bundled imports.
+          // declarations, resolve each PKG to its on-disk root directory, and
+          // strip the declaration from the source. Source files use this pattern
+          // when they need to reach internal files that the package's `exports`
+          // map doesn't expose; the next pass uses these roots to statically
+          // rewrite `require(`${NAME}/...`)` calls into bundled imports, so the
+          // declaration has no runtime purpose once those calls are rewritten.
+          // It MUST be removed: a surviving `require.resolve('PKG/package.json')`
+          // throws `Cannot find module` whenever the compiled binary runs from a
+          // directory tree that doesn't contain the package on disk (see #2282).
           const fileRequire = module.createRequire(args.path);
           const packageRoots = new Map<string, string>();
-          for (const [, varName, pkg] of source.matchAll(
+          source = source.replaceAll(
             /const\s+(\w+)\s*=\s*path\.dirname\(\s*require\.resolve\(\s*['"`]([^'"`]+)\/package\.json['"`]\s*\)\s*\)\s*;?/g,
-          )) {
-            packageRoots.set(varName, path.dirname(fileRequire.resolve(`${pkg}/package.json`)));
-          }
+            (_match, varName: string, pkg: string) => {
+              packageRoots.set(varName, path.dirname(fileRequire.resolve(`${pkg}/package.json`)));
+              return '';
+            },
+          );
 
           // Rewrite ``require(`${ROOT}/subpath`)`` template literals whose
           // ROOT was discovered above. Other dynamic requires (unknown var,
@@ -221,26 +230,40 @@ if (argv.platform === 'darwin') {
 
 logger.info(`Output: ${FsUtil.sizeReadable(await FsUtil.size(output))}`);
 
-logger.info(`Testing: '${output}' ...`);
-const procOutput = await new Promise<string>((resolve, reject) => {
-  const proc = child_process.spawn(output, ['--help'], { windowsHide: true });
-  let procOutput = '';
-  proc.stdout.on('data', (chunk: Buffer) => {
-    procOutput += chunk.toString();
+// Run from an isolated directory that contains no source files
+const testDirectory = await FsUtil.mkdtemp(path.join(Temp.getTempDir(), 'compile'));
+try {
+  const testFile = path.join(testDirectory, path.basename(output));
+  logger.info(`Copying: '${output}' -> '${testFile}' ...`);
+  await FsUtil.copyFile(output, testFile);
+  await fs.promises.chmod(testFile, 0o755); // chmod +x
+
+  logger.info(`Testing: '${testFile}' ...`);
+  const procOutput = await new Promise<string>((resolve, reject) => {
+    const proc = child_process.spawn(testFile, ['--help'], {
+      cwd: path.dirname(testFile),
+      windowsHide: true,
+    });
+    let procOutput = '';
+    proc.stdout.on('data', (chunk: Buffer) => {
+      procOutput += chunk.toString();
+    });
+    proc.stderr.on('data', (chunk: Buffer) => {
+      procOutput += chunk.toString();
+    });
+    proc.on('close', (code) => {
+      if (code === 0) {
+        resolve(procOutput);
+      } else {
+        reject(new Error(`${testFile} exited with code ${code}`));
+      }
+    });
+    proc.on('error', reject);
   });
-  proc.stderr.on('data', (chunk: Buffer) => {
-    procOutput += chunk.toString();
-  });
-  proc.on('close', (code) => {
-    if (code === 0) {
-      resolve(procOutput);
-    } else {
-      reject(new Error(`${output} exited with code ${code}`));
-    }
-  });
-  proc.on('error', reject);
-});
-logger.trace(procOutput);
+  logger.trace(procOutput);
+} finally {
+  await FsUtil.rm(testDirectory, { recursive: true, force: true });
+}
 
 Timer.cancelAll();
 logger.info('Finished!');

--- a/scripts/compile.ts
+++ b/scripts/compile.ts
@@ -3,7 +3,6 @@
 import child_process from 'node:child_process';
 import fs from 'node:fs';
 import module from 'node:module';
-import os from 'node:os';
 import path from 'node:path';
 
 import fg from 'fast-glob';

